osu.Game/Screens/Select/SongSelect.cs

```diff
@@ -253,8 +253,6 @@ private void carouselSelectionChanged(BeatmapInfo beatmap)
             }
             else
             {
-                Ruleset.Value = beatmap.Ruleset;

```

Comment:
I thought so too, so I looked through the code first to see what it would affect.  It's a little hard without knowing the whole framework yet.  It still correctly selects the game mode when you start playing, though.